### PR TITLE
LPS-156305 Update ObjectFieldLocalServiceTest to test defaultValue and state

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:forward

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 1]
```diff
@@ -1104,6 +1272,54 @@ private String _testAddSystemObjectField(
...
+				expectedObjectField.isIndexedAsKeyword(),
+				expectedObjectField.getIndexedLanguageId(),
+				expectedObjectField.getLabelMap(),
+				expectedObjectField.getName(), expectedObjectField.isRequired(),
+				expectedObjectField.isState(),
+				expectedObjectField.getObjectFieldSettings(), null);
+
+		Assert.assertEquals(
+			expectedObjectField.getExternalReferenceCode(),
+			objectField.getExternalReferenceCode());
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1296]: Hi @seltong,

The externalReferenceCode assert should be after of the defaultValue assert.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 2]
```diff
@@ -119,6 +147,100 @@ public void testAddCustomObjectField() throws Exception {
...
+					true
+				).build());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldStateException objectFieldStateException) {
+			Assert.assertEquals(
+				"Object field must be required when the state is true",
+				objectFieldStateException.getMessage());
+		}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:238]: Hi @seltong,

Lets try to apply something like [this method](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectViewLocalServiceTest.java#L400-L432), I believe the that case is equals to this.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:relevant

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java [hunk 3]
```diff
@@ -982,6 +1112,29 @@ public void testUpdateCustomObjectField() throws Exception {
...
+		try {
+			_testAddCustomObjectField(objectField);
+
+			Assert.fail();
+		}
+		catch (PortalException portalException) {
+			if (!expectedExceptionClass.isInstance(portalException)) {
+				Assert.fail();
+			}
+
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java:1135]: Hi @seltong,

Could you delete this condition above, thx!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:forward

**Pull request closed without merging**